### ISSUE TITLE
Psalm uses PHPStan's @var annotation when it's a union type

### ISSUE THREAD

**Issue by MidnightDesign** [vimeo/psalm#5807]
https://psalm.dev/r/5ab480345a

```php
<?php

class Foo {}

class Bar {
    /**
     * @phpstan-var Foo<string>|null
     * @psalm-var Foo|null
     */
    private ?Foo $foo = null;
}
```

Expected:
No issues reported because `@psalm-var` doesn't use a type argument for `Foo`.

Actual:
> ERROR: TooManyTemplateParams - 10:5 - Foo<string> has too many template params, expecting 0

Additional info:
- [It doesn't report an issue when the `@phpstan-var` annotation is removed](https://psalm.dev/r/0c170c94b0)
- [It doesn't report an issue when the type is not a union](https://psalm.dev/r/869fdd8cda)

**Comment by orklah** [comment]
It reports the issue even when the template is in a @var instead of @phpstan-var: https://psalm.dev/r/b9267465b3
